cpp/src/table/row_operators.cu

```diff
@@ -317,10 +326,12 @@ void check_lex_compatibility(table_view const& input)
     if (c.type().id() == type_id::LIST) {
       auto const& list_col = lists_column_view(c);
       CUDF_EXPECTS(list_col.child().type().id() != type_id::STRUCT,
-                   "Cannot lexicographic compare a table with a LIST of STRUCT column");
+                   "Cannot lexicographically compare a table with a LIST of STRUCT column");
```

Comment:
We can now compare list of struct column, right?

Reply:
This check is to verify the intermediate result, which still can't contain lists-of-structs or structs-of-lists.

The input can be anything and will be supported all. But it will be preprocessed to not contain lists-of-structs or structs-of-lists.

Reply:
> This error is basically for the case where calling code fails to do the necessary preprocessing before calling this function, right?

Exactly.